src/CitizenProject/CitizenProjectManager.php

```diff
@@ -498,14 +498,54 @@ public function hasCitizenProjectInStatus(Adherent $adherent, array $status): bo
         return $this->getCitizenProjectRepository()->hasCitizenProjectInStatus($adherent, $status);
     }
 
+    /**
+     * Uploads and saves the citizen project image.
+     *
+     * @param CitizenProject $citizenProject
+     */
     public function addImage(CitizenProject $citizenProject): void
     {
-        // Save citizen project image to cloud storage
-        if ($citizenProject->getImage() instanceof UploadedFile) {
-            $pathImage = $citizenProject->getImagePath();
-            $this->storage->put($pathImage, file_get_contents($citizenProject->getImage()->getPathname()));
-            $this->glide->deleteCache($pathImage);
-            $citizenProject->setImageUploaded(true);
+        if (!$citizenProject->getImage() instanceof UploadedFile) {
+            throw new \RuntimeException(sprintf('The image must be an instance of %s', UploadedFile::class));
```

Comment:
Can't it be null? What about submitting a form without editing or removing the image, in such case null should be allowed, right?

Reply:
If the image is not submitted, the `CitizenProjectCreationCommandHandler` and the `CitizenProjectUpdateCommandHandler`  not call this `addImage` method : 
```        // Uploads an image
        if (null !== $command->getImage()) {
            $this->citizenProjectManager->addImage($citizenProject);
        } 
```

Reply:
Fait enough